plugins/modules/panos_readiness_checks.py

```diff
@@ -0,0 +1,195 @@
+        type: bool
+        default: false
+"""
+
+EXAMPLES = """
+- name: Run all management plane checks using NOT notation
+  panos_readiness_checks:
+    provider: '{{ device }}'
+    checks:
+      - '!ha'
+      - '!session_exist'
+      - '!arp_entry_exist'
+      - '!ip_sec_tunnel_status'
+
+- name: Check if a specified session exists in vsys2, fail if it does not
+  panos_readiness_checks:
+    provider: '{{ device }}'
+    vsys: vsys2
+    force_fail: true
+    checks:
+      - session_exist:
+          source: '34.23.15.1'
+          destination: '10.1.0.4'
+          dest_port: '80'
+"""
+
+RETURN = """
+response:
+    description:
+        - This is a B(dict) where keys are checks names just as you specify them in the I(checks) property.
+        - Each value is also a B(dict).
+        - WHen I(force_fail) has the default value of B(false) this B(dict) contains results for all checks that were specified in I(checks) property.
+        - When I(force_fail) is set to B(true) it contains only checks that failed.
+    type: dict
+    returned: always
+    sample:
+        arp_entry_exist:
+            reason: "[SKIPPED] Missing ARP table entry description."
+            state: false
+        candidate_config:
+            reason: "[FAIL] Pending changes found on device."
+            state: false
+        content_version:
+            reason: "[FAIL] Installed content DB version (8640-7694) is not the latest one (8697-7981)."
+            state: false
+        free_disk_space:
+            reason: "[SUCCESS] "
+            state: true
+        ha:
+            reason: "[ERROR] Device is not a member of an HA pair."
+            state: false
+        ip_sec_tunnel_status:
+            reason: "[SKIPPED] Missing tunnel specification."
+            state: false
+        ntp_sync:
+            reason: "[ERROR] No NTP server configured."
+            state: false
+        panorama:
+            reason: "[SUCCESS] "
+            state: true
+        session_exist:
+            reason: "[SKIPPED] Missing critical session description. Failing check."
+            state: false
+    contains:
+        state:
+            description: A result of a check.
+            type: bool
+            returned: always
+        reason:
+            description:
+                - A free text describing the check result.
+                - 'Prefixed with a keyword: SUCCESS, FAIL, ERROR, SKIPPED.'
+                - Meaningful only for failed tests as the ones succeeded are self explanatory.
+            type: str
+            returned: always
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.paloaltonetworks.panos.plugins.module_utils.panos import (
+    get_connection,
+)
+
+MIN_PUA_VER = (0, 3, 0)
+
+try:
+    from panos_upgrade_assurance.check_firewall import CheckFirewall
+    from panos_upgrade_assurance.firewall_proxy import FirewallProxy
+except ImportError:
+    pass
```

Comment:
Why don't we have the `PUA_AVAILABLE` logic here as in the `panos_snapshot_report.py`?

Reply:
same here